pkg/metrics/gameservers.go

```diff
@@ -0,0 +1,72 @@
+// Copyright 2018 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package metrics
+
+import (
+	"context"
+
+	stablev1alpha1 "agones.dev/agones/pkg/apis/stable/v1alpha1"
+	"go.opencensus.io/stats"
+	"go.opencensus.io/tag"
+	"k8s.io/apimachinery/pkg/util/errors"
+)
+
+// GameServerCount  is the count of gameserver per current state and per fleet name
+type GameServerCount map[stablev1alpha1.State]map[string]int64
+
+// increment adds the count of gameservers for a given fleetName and state
+func (c GameServerCount) increment(fleetName string, state stablev1alpha1.State) {
+	fleets, ok := c[state]
+	if !ok {
+		fleets = map[string]int64{}
+		c[state] = fleets
+	}
+	fleets[fleetName]++
+}
+
+// reset sets zero to the whole metrics set
+func (c GameServerCount) reset() {
+	for _, fleets := range c {
+		for fleet := range fleets {
+			fleets[fleet] = 0
+		}
+	}
+}
+
+// record counts the list of gameserver per status and fleet name and record it to OpenCensus
+func (c GameServerCount) record(gameservers []*stablev1alpha1.GameServer) error {
+	// Currently there is no way to remove a metric so we have to reset our values to zero
```

Comment:
@markmandel Can we check with OpenCensus team if there is a way to tackle this ? I found this annoying but I understand why it happens. 

With Prometheus not recording a metrics with tags at scrape time, considered  it deleted/missing.

May be we could have a way to reset a gauge ? or deleted it ?

Reply:
Ok I’ll report it, FYI I’m counting gameservers per fleet name and status in a map, then going through the map to record each results, the problem is that if there is no more gameservers for a given fleet I have to reset the value to zero but the stats stays, so it’s ever growing, may be unregister/register the view each time.

Reply:
I figure asking on the Gitter is the best bet - otherwise, I've no idea what I'm asking :smile:

Reply:
went for an issue instead - https://github.com/census-instrumentation/opencensus-go/issues/1003

I think it's easier to keep track of.

Reply:
Do you want to block on this until we get an answer from the OpenCensus team, or shall we iterate as we get more information?

Reply:
I don’t expect a solution by 0.7 so let’s go with what we have.